ghga_event_schemas/pydantic_.py

```diff
@@ -272,13 +284,19 @@ class Config:
 
 class NonStagedFileRequested(BaseModel):
     """
-    This event type is triggered when a user requested to download a file that is not
-    yet present in the outbox and need to be staged.
+    This event type is triggered when a user requests to download a file that is not
+    yet present in the outbox and needs to be staged.
     """
 
     file_id: str = Field(
         ..., description="The public ID of the file as present in the metadata catalog."
     )
+    object_id: str = Field(
+        ..., description="The ID of the file in the specific S3 bucket."
+    )
+    bucket_id: str = Field(
+        ..., description="The ID/name of the S3 bucket used to store the file."
+    )
```

Comment:
The object ID is not yet known but chosen by the internal file registry service upon copy. The download controller just instructs the internal file registry which bucket to use:
```suggestion
    object_id: str = Field(
        ..., description="The ID of the file in the specific S3 bucket."
    )
    target_bucket_id: str = Field(
        ..., description="The ID/name of the S3 bucket in which the object was expected."
    )
```

Reply:
If we want to name it that way, then the other events should probably refer to it as source_bucket_id

Reply:
Yes sounds good. We can call it source_bucket_id and source_object_id

Reply:
Oh I see it is inheriting. But let's clarify who is setting the object id and whether the Staged event is needed at all.